debsources/app/patches/views.py

```diff
@@ -73,41 +108,23 @@ def parse_patch_series(self, session, package, version, config, series):
                                                bug='')
         return patches_info
 
-    def get_objects(self, path_to):
-        path_dict = path_to.split('/')
-        package = path_dict[0]
-        version = path_dict[1]
+    def get_objects(self, packagename, version):
+        path_to = packagename + '/' + version
 
-        if len(path_dict) > 2:
-            raise Http404ErrorSuggestions(package, version, '')
-
-        if version == "latest":  # we search the latest available version
-            return self._handle_latest_version(request.endpoint,
-                                               package, "")
-
-        versions = self.handle_versions(version, package, "")
-        if versions:
-            redirect_url_parts = [package, versions[-1]]
-            redirect_url = '/'.join(redirect_url_parts)
-            return self._redirect_to_url(request.endpoint,
-                                         redirect_url, redirect_code=302)
-
-        # identify patch format, accept only 3.0 quilt
         try:
-            source_format, loc = get_sources_path(session, package, version,
-                                                  current_app.config,
-                                                  'debian/source/format')
-        except (FileOrFolderNotFound, InvalidPackageOrVersionError):
-            return dict(package=package,
-                        version=version,
-                        path=path_to,
-                        patches=[],
-                        format='unknown')
-
-        with io.open(source_format, mode='r', encoding='utf-8') as f:
-            format_file = f.read()
-        if format_file.rstrip() not in ACCEPTED_FORMATS:
-            return dict(package=package,
+            format_file = helper.get_patch_format(session, packagename,
+                                                  version, current_app.config)
+        except (FileOrFolderNotFound, InvalidPackageOrVersionError) as e:
+            if isinstance(e, InvalidPackageOrVersionError):
```

Comment:
please use 2 different "except" here, instead of checking the type of the exception